### DISCUSSION TITLE
end of tube note woes

### DISCUSSION THREAD

**Discussion by alexDiatonic** [edwardkort/WWIDesigner#145] [Flute Study]
Hi,

I am hoping for some help, I am modelling a professionally-made keyless D flute with a C natural hole which I own. The aim is to use this exercise to get an understanding of WID so that I can use this incredibly promising tool to design a flute from scratch.

The predicted frequencies are looking close enough to the measured frequencies except the second octave D, first finger off, which is much flatter than measured (predicted -29.4 cents, measured at -2). Also it predicts 11.2 cents sharper with the first finger on, which seems strange, and isn't what the real flute does. The low D prediction is accurate at +4.4 cents.

Any ideas why this might be?

I calibrated  using the measured freqs of low d and its overtones, all fingers on, as I would when building a flute and before drilling tone holes. This gave a beta of 0.2 airstream length of 2.8mm. I am going with the recommended 1.5mm for airstream height.

I have noticed the 'end of the tube' note seems to be a special case in being very sensitive to calibration factors, 'beta' and especially to 'airstream height'. Is that where the answer might lie?

I wouldn't rule out a typo. measurement error or some undetected subtly of the bore, although I have tried to check everything carefully.

Alex

[WI designer model.zip](https://github.com/edwardkort/WWIDesigner/files/8102140/WI.designer.model.zip)

**Comment by bhp1** [comment]
The default constraints for the flute calibrator limit beta to a minimum of 0.2.  This suggests the model isn't doing a good job of modelling the real flute.  Because you have a flute with all of its holes, I suggest you calibrate the model with all of the notes.  If you can, measure an actual playing frequency for every note over the full two octaves, maintaining an even increase in breath speed as you go from the D4 through D6.  Avoid trying to blow the notes in tune.  Put the measured frequencies in the "Frequency" column of a separate tuning file, and use that for calibration.  In the options, set the blowing level to 9 or 10; that has given good results with at least one other real flute.

Meanwhile, I'll spend some time with the files you sent to see what else might be going on.  I will say that flutes are tougher to model than whistles, because the player has more scope for adjustment.

Is the thumb hole kept closed for all notes except C-nat?  I notice your tuning file has it open for the two C# notes.

**Reply by alexDiatonic** [reply]
Thanks a lot for your help,
Yes, I do lift my thumb for C natural, I'm not sure if what the maker intended, but it follows from my Boehem instincts I guess. 

I tried running the calibration with all notes, frequencies measured with Flutini and trying not to correct pitches, the effect was the same, giving a beta of 0.2 and the same anomaly in the tuning. Also I added some extra bore measurements to fill gaps, located a few minor changes in the taper, but the predicted tuning comes out with the same general picture, although I will now double check the whole lot again to be sure.

Here is something I noticed that may or may not help to identify what is going on, maybe this is expected: 
take the instrument file as uploaded, change the beta to 0.4, then lengthen the head joint by 10mm (subtract 10 from stopper and mouthpiece position) and check tuning. The second octave D drops by more than 350 cents. while all the other notes move in a the range of 12-50 cents. 

Alex

**Comment by bhp1** [comment]
The tuning file you sent had weights of zero for the top three notes, C6, C#6, and D6.  Be sure to set those weights to 1 when you run the calibration.  If I pretend the flute plays a perfect two octave scale (maybe not far from the truth, given the maker) and calibrate it against the ET scale, I get an airstream length around 2.9 mm and beta around 0.29... much more reasonable values.

The model still predicts D5 will be around 24 cents flat.  Look at [the tuning graph](https://github.com/edwardkort/WWIDesigner/wiki/Evaluating-Instruments-with-the-Tuning-Table-and-Tuning-Graph#the-tuning-graph).  For C5, C#5, and E5, the blue circle (the predicted playing frequency on the breath curve) is very close to the green dot (the in-tune note), but for D5, the green dot is higher than the blue circle.  If you play an in-tune C#5, then switch to D5 with (almost) no change in air speed, is the D5 in tune or flat?  Similarly, if you play an in-tune E5 and switch to D5 with no change in air speed, is the D5 in tune or flat?  

If those D5s are in tune, then there is something wrong with how WIDesigner models that fingering on that instrument.  Maybe we can get around this by fudging something in the instrument file.

If those D5s are flat, but D5s are just fine when you are playing them in a tune, then there is something about playing style, how you approach that note in real life, that WIDesigner's model doesn't allow for.  In this case, we might need a way to customize the breath curve to accommodate this.  See [discussion #128](https://github.com/edwardkort/WWIDesigner/discussions/128).  Without customized breath curves, you can still use the tuning graph to get a sense of how well the pattern of green dots matches how you blow the notes in practice.

**Reply by alexDiatonic** [reply]
Thanks again for your help,

I think what I will do is at this point, is carefully recheck all the geometry as well as the intonation and report back.

More generally, when designing a flute from scratch, since this is the ultimate aim, how would I choose a beta value to test the design against? Would you expect to be able to correct for the range of likely beta values with tweaks to hole size and spacing?

**Reply by bhp1** [reply]
I think your first step is a good one: model yourself playing a known, finished flute.  Another useful step would be modelling yourself playing a flute (or any flute-like object) with a headjoint of your own design.

We expect beta to be around 0.3-0.4; maybe 0.2 to 0.5 at the outside limits.  If all else fails, you can start with 0.3 or 0.4, and refine the value as you get experience with physical objects closer to your final design.

**Comment by alexDiatonic** [comment]
OK, so I went through the geometry again, and tried recalibrating with a range of head joint positions, changing how far the flute was rolled in and out and how far the tenon extended. The software took me through a range of calibration values but none of them was satisfactory. 

It seems like there is some kind of gremlin in the model that makes it extremely sensitive to small changes, as well as not predicting the tuning well. I couldn't find the gremlin amongst my inputs, but I definitely wouldn't rule that out with so many values entered.  

So I went to plan b-  set aside the wooden flute and started measuring up my Aulos AF2 baroque flute in ABS plastic. I got an excellent result with this. The predicted tuning was almost all within 10c of measured. I also tried a bit of a sensitivity test, making a series of arbitrary changes to geometry and calibration values. Reassuringly the affect of these changes on the predicted tuning all seemed in the right direction and within a reasonable magnitude.

So I'm moving on from this exercise and on to planning one to build, and I am fairly confident WIDesigner will be helpful for me. Thanks a lot for your assistance and for creating and making this software available.

**Comment by bhp1** [comment]
The general opinion of the flute's maker tells me it isn't the flute.  Your experience with the Aulos suggests it isn't you.  That means the problem is in WIDesigner.  I'd really like to know where the model is going wrong, or what it fails to capture about that flute.

Since it is a hand-crafted flute, is it possible that the maker has made fine adjustments to tune specific notes, such as an angled tonehole, or undercutting the inside of a tonehole?

**Reply by alexDiatonic** [reply]
I can't see any significant undercutting or angled holes, except the undercutting on the embouchure hole. I don't know a way to actually measure undercutting however. 

The only feature that surprised me is that there is a drop of diameter from the end of the head joint (not counting the socket of course) and the tip of the tenon of the middle section, I measured as 0.7mm. I entered this as two points 0.1mm apart. I don't know if this is unusual, this suggests probably not http://www.mcgee-flutes.com/bores.htm.

This 'top of the taper' area seems like the where something is going wrong -  I can't get past the prediction that D5 will get flatter with the first finger off. It is not what the real flute does, but, from principles is this acoustically plausible? It seems counter intuitive to me. A survey of flutes and whistles I have around shows lifting the finger either doesn't change the pitch or raises it.

If I create a simplified four point (three section) version of the bore with similar holes it has the same property, so the effect is not emerging from the minor lumps and bumps. This is attached in case its handy.

I suppose everything in the model is interdependent so there no way to pinpoint a cause of a particular pitch prediction
[WI.zip](https://github.com/edwardkort/WWIDesigner/files/8262351/WI.zip)
.

**Comment by bhp1** [comment]
I checked the predictions for vented and unvented fingerings for D5, using the attached tuning file.  Note that all the weights are 1, so the high D6 is included in the scale; this is important because it affects the predictions for all the other notes (because it determines where we put the top end of the breath curve).  The prediction for vented D5 (top finger off) is less than 2 Hz lower than unvented D5,  I'd say this is less than the uncertainty in the prediction model, so I wouldn't put too much emphasis on it.  (I tried a different flute model, and the vented D5 came out about the same amount higher than unvented D5.)

Vented or unvented, the prediction for D5 is still around 20-25 cents flat.  That does bug me.

[Tuning-D4-Equal-withCnatHole, 2D5.zip](https://github.com/edwardkort/WWIDesigner/files/8279750/Tuning-D4-Equal-withCnatHole.2D5.zip)
